File: javascript/bob/bob.js
export const hey = (message) => {
  const FORCEFUL_QUESTION = /^(?<=\b)[^a-z0-9]+\?$/
  const REGULAR_QUESTION = /\?$/
  const YELLING = /^[^a-z]+(?<!\d)$/
  const SILENCE = /^\s*$/

  message = message.trim();

  if (message.match(FORCEFUL_QUESTION)) return "Calm down, I know what I'm doing!"
  else if (message.match(REGULAR_QUESTION)) return "Sure."
  else if (message.match(YELLING)) return "Whoa, chill out!"
  else if (message.match(SILENCE)) return "Fine. Be that way!"
  else return "Whatever."
};

File: ruby/bob/bob.rb
class Bob
  FORCEFUL_QUESTION = /\A(?<=\b)[^a-z0-9]+\?\z/
  REGULAR_QUESTION = /\?\z/
  YELLING = /\A[^a-z]+(?<!\d)\z/
  SILENCE = /\A\s*\z/

  def self.hey(remark)
    case remark.strip
    when FORCEFUL_QUESTION then "Calm down, I know what I'm doing!"
    when REGULAR_QUESTION then "Sure."
    when YELLING then "Whoa, chill out!"
    when SILENCE then "Fine. Be that way!"
    else "Whatever."
    end
  end
end
